pkg/apis/core/types_shoot.go

```diff
@@ -556,6 +556,21 @@ type KubeletConfig struct {
 	ImagePullProgressDeadline *metav1.Duration
 	// FailSwapOn makes the Kubelet fail to start if swap is enabled on the node. (default true).
 	FailSwapOn *bool
+	// KubeReserved is the configuration for resources reserved for kubernetes node components.
+	// Default:
+	//   memory:	1Gi
+	//   cpu:		80m
+	KubeReserved *KubeletConfigKubeReserved
+	// SystemReserved is the configuration for resources reserved for system components.
+	SystemReserved *KubeletConfigSystemReserved
+	// EnforceNodeAllocatable is an array of cgroups whose resource would be enforced by kubelet.
+	// Default:
+	//   - pods
+	EnforceNodeAllocatable []string
+	// KubeReservedCgroup is the name of the cgroup that kubelet will enforce kube-reserved resource on.
+	KubeReservedCgroup *string
+	// SystemReservedCgroup is the name of the cgroup that kubelet will enforce system-reserved resource on.
```

Comment:
// The configured cgroup is not automatically created and thus must exist on the worker node before using this option.